crates/core/src/delta_datafusion/expr.rs

```diff
@@ -339,6 +343,36 @@ impl<'a> fmt::Display for ScalarValueFormat<'a> {
             ScalarValue::UInt16(e) => format_option!(f, e)?,
             ScalarValue::UInt32(e) => format_option!(f, e)?,
             ScalarValue::UInt64(e) => format_option!(f, e)?,
+            ScalarValue::Date32(e) => match e {
+                Some(e) => {
+                    let dt =
+                        NaiveDate::from_num_days_from_ce_opt((EPOCH_DAYS_FROM_CE + (*e)).into())
+                            .unwrap();
+                    write!(f, "{}", dt)?
+                }
+                None => write!(f, "NULL")?,
+            },
+            ScalarValue::Date64(e) => match e {
+                Some(e) => write!(
+                    f,
+                    "{}",
+                    NaiveDateTime::from_timestamp_millis((*e).into())
+                        .unwrap()
+                        .date()
+                )?,
+                None => write!(f, "NULL")?,
+            },
+            ScalarValue::TimestampMicrosecond(e, tz) => match e {
+                Some(e) => match tz {
+                    Some(tz) => write!(
+                        f,
+                        "{}",
+                        NaiveDateTime::from_timestamp_micros(*e).unwrap().and_utc()
+                    )?,
+                    None => write!(f, "{}", NaiveDateTime::from_timestamp_micros(*e).unwrap())?,
```

Comment:
and finally here :-)

Reply:
@hntd187 it actually is not an error but it returns Option<NaiveDateTime> when the value that is passed is out of range.. This generally shouldn't happen but I can return an error when it's None though.